test/e2e/common/util_cleanup.go

```diff
@@ -211,11 +294,58 @@ func ForceDeleteMayastorPods() bool {
 		}
 	}
 
+	podCount := 0
 	// We have made the best effort to cleanup, give things time to settle.
-	for attempts := 0; attempts < 30 && MayastorUndeletedPodCount() != 0; attempts++ {
+	for attempts := 0; attempts < 60 && MayastorUndeletedPodCount() != 0; attempts++ {
+		pods, err := gTestEnv.KubeInt.CoreV1().Pods("mayastor").List(context.TODO(), metav1.ListOptions{})
+		if err == nil {
+			podCount = len(pods.Items); if podCount == 0 {
+				break
+			}
+		}
 		time.Sleep(2 * time.Second)
 	}
 
-	logf.Log.Info("EnsureMayastorDeleted: lingering Mayastor pods were found !!!!!!!!")
-	return true
+	return true, podCount, nil
+}
+
+// "Big" sweep, attempts to remove artefacts left over in the cluster
+// that would prevent future successful test runs.
+func CleanUp() (bool) {
+	podCount, delPodsErr := DeleteAllPods()
+	pvcCount, delPvcErr := DeleteAllPvcs()
+	pvCount, delPvErr := DeleteAllPvs()
+	msvCount, delMsvErr := DeleteAllMsvs()
+	// Pools should not have finalizers if there are no associated volume resources.
+	poolFinalizerDeleted, delPoolFinalizeErr := DeleteAllPoolFinalizers()
+
+	logf.Log.Info("Resource cleanup",
+		"podCount", podCount,
+		"pvcCount", pvcCount,
+		"pvCount", pvCount,
+		"msvCount", msvCount,
+		"delPodsErr", delPodsErr,
+		"delPvcErr", delPvcErr,
+		"delPvErr", delPvErr,
+		"delMsvErr", delMsvErr,
+		"poolFinalizerDeleted", poolFinalizerDeleted,
+		"delPoolFinalizeErr", delPoolFinalizeErr,
+	)
+
+	scList, delScErr := gTestEnv.KubeInt.StorageV1().StorageClasses().List(context.TODO(), metav1.ListOptions{})
+	if delScErr == nil {
+		for _, sc := range scList.Items {
+			if sc.Provisioner == "io.openebs.csi-mayastor" && sc.Name != "mayastor-iscsi" && sc.Name != "mayastor-nvmf" {
+				logf.Log.Info("Deleting", "storageClass", sc.Name)
+				_ = gTestEnv.KubeInt.StorageV1().StorageClasses().Delete(context.TODO(), sc.Name, metav1.DeleteOptions{GracePeriodSeconds: &ZeroInt64})
+			}
+		}
+	} else {
+		logf.Log.Info("Error listing storage classes", "error", delScErr)
```

Comment:
Should this be an Error?

Reply:
AFAICT using Error gives us a backtraces, which are only useful for failures intrinsic to the test as opposed to entities being tested. Which includes the cluster IMO.